src/context/ThemeContext.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import { Theme } from "./ThemeContext.types";
+import { ThemeContextType } from "./ThemeContext.types";
+
+const defaultTheme: Theme =
+  window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches
+    ? "dark"
+    : "light";
+
+const ThemeContext = createContext<ThemeContextType>({
+  theme: defaultTheme,
+  toggleTheme: () => {},
+});
+
+export const ThemeProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [theme, setTheme] = useState<Theme>(
+    (localStorage.getItem("theme") as Theme) || defaultTheme,
+  );
+
+  useEffect(() => {
+    localStorage.setItem("theme", theme);
+    document.documentElement.className = theme;
```

Comment:
It's quite unsafety to define className, as it will delete previous state completely and define only one class. Try to image the situation there is a class from third party library and you remove it.